src/util/serializing/ObjectInputStream.cpp

```diff
@@ -94,94 +76,46 @@ auto ObjectInputStream::readString() -> std::string {
 
     size_t lenString = (size_t)readTypeFromSStream<int>(istream);
 
-    if (istream.str().size() < len) {
-        throw InputStreamException("End reached, but try to read an string", __FILE__, __LINE__);
-    }
-
     std::string output;
     output.resize(lenString);
 
     istream.read(&output[0], (long)lenString);
-
-    return output;
-}
-
-void ObjectInputStream::readData(void** data, int* length) {
-    checkType('b');
-
-    if (istream.str().size() < 2 * sizeof(int)) {
-        throw InputStreamException("End reached, but try to read data len and width", __FILE__, __LINE__);
-    }
-
-    int len = readTypeFromSStream<int>(istream);
-    int width = readTypeFromSStream<int>(istream);
-
-    if (istream.str().size() < static_cast<size_t>(len * width)) {
-        throw InputStreamException("End reached, but try to read data", __FILE__, __LINE__);
-    }
-
-    if (len == 0) {
-        *length = 0;
-        *data = nullptr;
-    } else {
-        *data = (void*)new char[(size_t)(len * width)];
-        *length = len;
-
-        istream.read((char*)*data, len * width);
-    }
-}
-
-template <typename T>
-void ObjectInputStream::readData(std::vector<T>& data) {
-    checkType('b');
-
-    if (istream.str().size() < 2 * sizeof(int)) {
-        throw InputStreamException("End reached, but try to read data len and width", __FILE__, __LINE__);
-    }
-
-    int len = readTypeFromSStream<int>(istream);
-    int width = readTypeFromSStream<int>(istream);
-
-    if (width != sizeof(T)) {
-        throw InputStreamException("Data width mismatch requested type width", __FILE__, __LINE__);
+    if (istream.fail()) {
+        throw InputStreamException("End reached, but try to read an string", __FILE__, __LINE__);
     }
 
-    if (istream.str().size() < static_cast<size_t>(len * width)) {
-        throw InputStreamException("End reached, but try to read data", __FILE__, __LINE__);
-    }
 
-    if (len) {
-        data.resize(len);
-        istream.read((char*)data.data(), len * width);
-    }
+    return output;
 }
 
-auto ObjectInputStream::readImage() -> std::string {
+auto ObjectInputStream::readImage() -> const std::vector<std::byte> {
```

Comment:
```suggestion
auto ObjectInputStream::readImage() -> std::vector<std::byte> {
```

Reply:
If the data must not be edited, you should return std::vector<std::byte const>, but that can result in undefined behav, when the original vector is also defined in that way and someone casts away the const qualifier:


```
1. std::vector<std::byte const> vec;
2. return vec;
3. reinterpret_cast<char*> (vec.data()); // undefined behav.
3. reinterpret_cast<char const*> (vec.data()); // valid
```
vs:

```
1. std::vector<std::byte> vec;
2. return vec; // as std::vector<std::byte const> 
3. reinterpret_cast<char*> (vec.data()); // valid
3. reinterpret_cast<char const*> (vec.data()); // valid
```